src/lib.rs

```diff
@@ -111,3 +112,39 @@ where
         Ord::cmp(self, key.borrow())
     }
 }
+
+/// `ComparableRangeBounds` is implemented as an extention to `RangeBounds` to
+/// allow for comparison of items with range bounds.
+pub trait ComparableRangeBounds<T>: RangeBounds<T> {
+    /// Returns `true` if `item` is contained in the range.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// assert!( (3..5).comparable_contains(&4));
+    /// assert!(!(3..5).comparable_contains(&2));
+    ///
+    /// assert!( (0.0..1.0).comparable_contains(&0.5));
+    /// assert!(!(0.0..1.0).comparable_contains(&f32::NAN));
+    /// assert!(!(0.0..f32::NAN).comparable_contains(&0.5));
+    /// assert!(!(f32::NAN..1.0).comparable_contains(&0.5));
+    /// ```
+    fn comparable_contains<U>(&self, item: &U) -> bool
+    where
+        U: ?Sized + Comparable<T>,
```

Comment:
What is the intended use case? If it's something like `BTreeMap::range`, then I think the relationship should be the other way, `T: Comparable<U>`, or in map terms it's `RangeBounds<Q>` and `Q: Comparable<K>`.

Reply:
The intended use case is most like a `BTreeMap`, I need something like

```
fn foo<R, Q>(&self,  key: &K) 
where
  R: RangeBounds<Q>,
  Q: ?Sized + Comparable<K>
{
  if self.range.comparable_contains(key) { }
}
```